sdk/src/main/java/org/eclipse/velocitas/sdk/grpc/AsyncBrokerGrpcFacade.kt

```diff
@@ -0,0 +1,438 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.eclipse.velocitas.sdk.grpc
+
+import io.grpc.Channel
+import io.grpc.Context
+import io.grpc.stub.StreamObserver
+import org.eclipse.kuksa.proto.v2.KuksaValV2.*
+import org.eclipse.kuksa.proto.v2.Types.Datapoint
+import org.eclipse.kuksa.proto.v2.Types.SignalID
+import org.eclipse.kuksa.proto.v2.VALGrpc
+import org.eclipse.velocitas.sdk.Status
+import org.eclipse.velocitas.sdk.concurrency.AsyncResult
+
+/**
+ * AsyncBrokerGrpcFacade provides asynchronous communication against the VehicleDataBroker.
+ * The current implementation uses the 'kuksa.val.v2' protocol which can be found here: https://github.com/eclipse-kuksa/kuksa-databroker/tree/main/proto/kuksa/val/v1.
+ */
+class AsyncBrokerGrpcFacade(private val channel: Channel) : GrpcClient() {
+    private val asyncStub: VALGrpc.VALStub
+        get() = VALGrpc.newStub(channel)
+
+    /**
+     * Gets the latest value of a [signalId].
+     * The response is delivered as an AsyncResult which allows the response to be consumed asynchronously using an
+     * [AsyncResult.resultCallback] or synchronously by using [AsyncResult.await].
+     *
+     * The server might respond with the following GRPC error codes:
+     *    NOT_FOUND if the requested signal doesn't exist
+     *    PERMISSION_DENIED if access is denied
+     */
+    fun getValue(signalId: SignalID): AsyncResult<GetValueResponse> {
```

Comment:
I changed these calls to AsyncResult for a better user experience. User can now decide if he wants to call the corresponding gRPC method in a synchronous or asynchronous way. However, right now this has one drawback:
- the visibility of AsyncResult#insertResult and AsyncResult#insertError is public right now :(

Probably needs to be changed to internal but this will generate these ugly methods on java side (and will basically still be accessable from it)

I really hate that kotlin still does not plan to support package privacy